evm/evm.go

```diff
@@ -0,0 +1,36 @@
+package evm
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+var (
+	// MonomerGenesisRootHash is the known root hash of the monomer ethereum state trie after all predeployed contracts are created.
+	MonomerGenesisRootHash = common.HexToHash("0x4e171626bc7f95d0b053dab23c87327ee0266dda88b213a3e1f1357b741c0c35")
+	// TODO: is there a specific address we want for MonomerEVMOriginAddress?
+	// MonomerEVMOriginAddress is the address used for executing transactions in the monomer EVM.
+	MonomerEVMOriginAddress = common.HexToAddress("0xae76d9126c32d7eafe3f5b7bd5a7b44f2d5bb8b1")
+)
+
+func NewEVM(ethState vm.StateDB) (*vm.EVM, error) {
+	return vm.NewEVM(
+		vm.BlockContext{
+			BlockNumber: big.NewInt(0),
+			Transfer:    func(vm.StateDB, common.Address, common.Address, *uint256.Int) {},
+			GasLimit:    10000000,
+		},
+		vm.TxContext{
```

Comment:
Same, can we use `NewEVMTxContext`? Maybe you already tried this and found that filling in the structs manually is better (or significantly easier), but if it isn't much more effort, we should opt for safety and use the helpers.

Reply:
Good idea, I'll give it a try

Reply:
`NewEVMTxContext` just takes in a `Message` and pulls the origin address and gas price directly from the message. here's some additional checks for blobs but those won't matter for monomer. The message struct will still need to be created so I'll just opt to initialize the `TxContext` struct directly here.